cms/djangoapps/contentstore/rest_api/v1/views.py

```diff
@@ -58,45 +62,106 @@ def proctored_exam_settings(request, course_id):
+
+            # validate data formats and update the course module object
+            is_valid, errors, updated_data = CourseMetadata.validate_and_update_from_json(
+                course_module,
+                models_to_update,
+                user=request.user,
+            )
+
+            if not is_valid:
+                error_messages = [error.get('message') for error in errors]
                 return Response(
-                    'Course with course_id {} does not exist.'.format(course_id),
-                    status=status.HTTP_404_NOT_FOUND
+                    {'detail': error_messages},
+                    status=status.HTTP_400_BAD_REQUEST
                 )
 
-            course_metadata = CourseMetadata().fetch_all(course_module)
-            data = {}
-
-            # specify only the advanced settings we want to return
-            proctored_exam_settings_advanced_settings_keys = [
-                'enable_proctored_exams',
-                'allow_proctoring_opt_out',
-                'proctoring_provider',
-                'proctoring_escalation_email',
-                'create_zendesk_tickets',
-                'start'
-            ]
-            proctored_exam_settings_data = {
-                setting_key: setting_value.get('value')
-                for (setting_key, setting_value) in course_metadata.items()
-                if setting_key in proctored_exam_settings_advanced_settings_keys
-            }
-
-            data['proctored_exam_settings'] = proctored_exam_settings_data
-            data['available_proctoring_providers'] = get_available_providers()
-
-            # move start key:value out of proctored_exam_settings dictionary and change key
-            data['course_start_date'] = proctored_exam_settings_data['start']
-            del data['proctored_exam_settings']['start']
-
-            data['is_staff'] = request.user.is_staff
-
-            serializer = ProctoredExamConfigurationSerializer(data)
-
-            return Response(serializer.data)
+            # save to mongo
+            modulestore().update_item(course_module, request.user.id)
+
+            # merge updated settings with all existing settings.
+            # do this because fields that could not be modified are excluded from the result
+            course_metadata = {**course_metadata, **updated_data}
+            updated_setttings = self._get_proctored_exam_setting_values(course_metadata)
+            serializer = ProctoredExamSettingsSerializer(updated_setttings)
+            return Response({
+                'proctored_exam_settings': serializer.data
+            })
+
+    @classmethod
+    def _get_proctored_exam_setting_values(cls, course_metadata):
+        return {
+            setting_key: course_metadata[setting_key].get('value')
+            for setting_key in cls.PROCTORED_EXAM_SETTINGS_KEYS
+        }
+
+    @staticmethod
+    def _get_and_validate_course(user, course_id):
```

Comment:
I feel like this function obfuscates the fact that `get_course_and_check_access` checks whether the requesting user has the correct permissions to access the course in the Studio context. Is that what you mean by `validate`? `get_course_and_check_access` could raise a `403` in certain situations, and I don't think that the name `_get_and_validate_course` makes that clear; I would think that this function just checks for the existence of a course, not that the user also has access to it if it exists. Could we make this clearer? Right now, although this deduplicates some code, I think it ends up being a little more confusing.